src/nvim/terminal.c

```diff
@@ -169,6 +170,50 @@ static VTermScreenCallbacks vterm_screen_callbacks = {
   .sb_popline  = term_sb_pop,
 };
 
+static void emit_osc_event(void **argv)
+{
+  int command = (int)(ptrdiff_t)argv[0];
+  char *payload = argv[1];
+  int payload_length = (int)(ptrdiff_t)argv[2];
+
+  save_v_event_T save_v_event;
+  dict_T *dict = get_v_event(&save_v_event);
+
+  typval_T cmd;
+  cmd.v_type = VAR_NUMBER;
+  cmd.vval.v_number = command;
+  tv_dict_add_nr(dict, S_LEN("command"), cmd.vval.v_number);
+
+  typval_T str;
+  str.v_type = VAR_STRING;
+  str.vval.v_string = payload;
+  tv_dict_add_str_len(dict, S_LEN("payload"), str.vval.v_string, payload_length);
+
+  tv_dict_set_keys_readonly(dict);
+
+  apply_autocmds(EVENT_TERMOSC, NULL, NULL, false, curbuf);
```

Comment:
Would be useful to also include the terminal buffer number or channel number in the `data` dictionary to the autocommand callback.

Reply:
The buffer number is already included and it is easy to get the `'channel'` from that, so this isn't necessary.